Tests/TuistCoreTests/Xcode/XcodeControllerTests.swift

```diff
@@ -0,0 +1,47 @@
+import Foundation
+import XCTest
+
+@testable import TuistCore
+@testable import TuistCoreTesting
+
+final class XcodeControllerTests: XCTestCase {
+    var system: MockSystem!
+    var subject: XcodeController!
+    var fileHandler: MockFileHandler!
+
+    override func setUp() {
+        super.setUp()
+        system = MockSystem()
+        fileHandler = try! MockFileHandler()
+        subject = XcodeController(system: system)
+    }
+
+    func test_selected_when_xcodeSelectDoesntReturnThePath() throws {
+        // Given
+        system.errorCommand(["xcode-select", "-p"])
+
+        // When
+        let xcode = try subject.selected()
+
+        // Then
+        XCTAssertNil(xcode)
+    }
+
+    func test_selected_when_xcodeSelectReturnsThePath() throws {
+        // Given
+        let contentsPath = fileHandler.currentPath.appending(component: "Contents")
+        try fileHandler.createFolder(contentsPath)
+        let infoPlistPath = contentsPath.appending(component: "Info.plist")
+        let infoPlist = Xcode.InfoPlist(version: "3.2.1")
+        let infoPlistData = try PropertyListEncoder().encode(infoPlist)
+        try infoPlistData.write(to: infoPlistPath.url)
+
+        system.succeedCommand(["xcode-select", "-p"], output: infoPlistPath.pathString)
```

Comment:
Good catch. The test should use the `/Applications/Xcode.app/Contents/Developer/` path instead. The test passes with `/Applications/Xcode.app/Contents/Info.plist` because the implementation removes the last two components from the path, resulting in the Xcode bundle path. I've updated the test accordingly.